cmd/launcher/launcher.go

```diff
@@ -48,11 +48,9 @@ func runLauncher(ctx context.Context, cancel func(), opts *launcher.Options) err
 	// determine the root directory, create one if it's not provided
 	rootDirectory := opts.RootDirectory
 	if rootDirectory == "" {
-		rootDirectory = filepath.Join(os.TempDir(), defaultRootDirectory)
-		if _, err := os.Stat(rootDirectory); os.IsNotExist(err) {
-			if err := os.Mkdir(rootDirectory, fsutil.DirMode); err != nil {
-				return fmt.Errorf("creating temporary root directory: %w", err)
-			}
+		rootDirectory, err := agent.MkdirTemp(defaultRootDirectory)
```

Comment:
Worth noting the previous code created `rootDirectory` with `0755` permission, whereas `MkdirTemp` defaults to `0700`. However, we then call `os.MkdirAll` with `0700`. 🤷

Reply:
Related - the `os.MkdirAll` does not ensure the root directory's permissions are `0700`. Only if the directory does not exist beforehand.

Reply:
This path is only taken when `rootDirectory == ""` which is such an uncommon use case I just don't want to angst about it.

Reply:
Correct. Just to be clear if `rootDirectory` is non-empty and the dir exists, we will trust the dir even if it's permissions are wide open.

Reply:
I'm not sure that's wrong -- most of what we put into `rootDirectory` isn't private. I think when we create the private things we do set the mode.

Reply:
I think you're right. So far I haven't found anything which appears vulnerable to some sort of hijacking here.